packages/vow/src/when.js

```diff
@@ -1,51 +1,39 @@
 // @ts-check
-import { unwrapPromise, getVowPayload, basicE } from './vow-utils.js';
+import { getVowPayload, basicE } from './vow-utils.js';
 
 /**
- * @param {() => import('./types.js').VowKit<any>} makeVowKit
- * @param {(resolver: import('./types').VowResolver) => Promise<any>} providePromiseForVowResolver
  * @param {(reason: any) => boolean} [rejectionMeansRetry]
  */
-export const makeWhen = (
-  makeVowKit,
-  providePromiseForVowResolver,
-  rejectionMeansRetry = () => false,
-) => {
+export const makeWhen = (rejectionMeansRetry = () => false) => {
   /**
+   * Shorten any vows in `specimenP` until we achieve a final result.
+   *
    * @template T
    * @param {import('./types.js').ERef<T | import('./types.js').Vow<T>>} specimenP
    */
-  const when = specimenP => {
-    /** @type {import('./types.js').VowKit<T>} */
-    const { resolver } = makeVowKit();
-    const promise = providePromiseForVowResolver(resolver);
+  const when = async specimenP => {
```

Comment:
Another interface consistency question: shouldn't vow `when` have the same (awaitable, onFulfilled, onRejected) parameter signature as [`E.when`](https://github.com/endojs/endo/blob/7639f0ceb6f23dfdae5c925962c17a9368bac760/packages/eventual-send/src/E.js#L222-L237), rather than just (awaitable) like `Promise.resolve`?